H4LCppModule.py

```diff
@@ -338,10 +368,22 @@ def analyze(self, event):
             etaZ2_2j = self.worker.Z2_2j.Eta()
             pTZ2_2j = self.worker.Z2_2j.Pt()
             EneZ2_2j = self.worker.Z2_2j.E()
-
-
-        if (foundZZCandidate_4l or foundZZCandidate_2l2q):
-            #keepIt = True
+        
+        if (foundZZCandidate_2l2nu):
+            keepIt = True
+            passZZ2l2nuSelection = True
+            self.passZZ2l2nuEvts += 1
+        #     FatJet_PNZvsQCD = self.worker.FatJet_PNZvsQCD
+        #     self.out.fillBranch("FatJet_PNZvsQCD",FatJet_PNZvsQCD)
+            phiZ2_met = self.worker.Z2_met.Phi()
+            pTZ2_met = self.worker.Z2_met.Pt()
+            EneZ2_met = self.worker.Z2_met.E()
+            print("inside 2l2nu loop")
+            self.out.fillBranch("phiZ2_met",phiZ2_met)
+            self.out.fillBranch("pTZ2_met",pTZ2_met)
+            self.out.fillBranch("EneZ2_met",EneZ2_met)
```

Comment:
As we discussed last time, the fillBranch should be at the end. Not inside the if condition.